README.md

```diff
@@ -1,39 +1,597 @@
-# Daru::IO
+# Daru-IO
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/daru/io`. To experiment with that code, run `bin/console` for an interactive prompt.
+[![Build Status](https://travis-ci.org/athityakumar/daru-io.svg?branch=master)](https://travis-ci.org/athityakumar/daru-io)
+[![Yard Docs](http://img.shields.io/badge/yard-docs-blue.svg)](http://www.rubydoc.info/github/athityakumar/daru-io/master/)
+[![Inline docs](http://inch-ci.org/github/athityakumar/daru-io.png)](http://inch-ci.org/github/athityakumar/daru-io)
+[![Code Climate](https://codeclimate.com/github/athityakumar/daru-io.png)](https://codeclimate.com/github/athityakumar/daru-io)
+[![Stories in Ready](https://badge.waffle.io/athityakumar/daru-io.png?label=ready&title=Ready)](https://waffle.io/athityakumar/daru-io?utm_source=badge)
+[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
-TODO: Delete this and the text above, and describe your gem
+A Ruby plugin-gem to [daru gem](https://github.com/SciRuby/daru), that extends support for many Import and
+Export methods of **Daru::DataFrame**. This gem is intended to help Rubyists who are into Data Analysis or Web
+Development, by serving as a general purpose conversion library that takes input in one format (say, JSON) and
+converts it another format (say, Avro) while also making it incredibly easy to getting started on
+analyzing data with daru.
 
-## Installation
+While supporting various IO modules, daru-io also provides an easier way of adding more Importers / Exporters
+by means of monkey-patching. **It's strongly recommended to have a look at [this section](#tweaking-io-modules), if you're interested in tweaking and creating custom Importers / Exporters.**
```

Comment:
I don't think that mentioning of monkey-patching is appropriate here. In fact, we "patch" almost nothing. It is a framework for defining importers/exporters and (optionally) "plugging" them as `DataFrame` methods.

It is important, because "monkey-patching" is considered quite questionable practice in modern Ruby, and start the explanation with proud statement about monkey-patching is bad for gem's reputation.